### PR TITLE
Fixes to CLI and version reporting.

### PR DESCRIPTION
**Pull request by DarkDimius**
We used to report a hardcoded 0.0.1 version in one place and 2.11 in other places.
Now, we report consistently the version that was used in sbt.

Additionaly, if version countains SNAPSHOT or NIGHTLY it will also report git sha.

### CODE REVIEW

**Comment by liufengyun** [comment]
@DarkDimius Could you check the compilation failure and run CI again?

File: compiler/src/dotty/tools/dotc/config/Properties.scala [hunk 1]
```diff
@@ -135,4 +135,10 @@ trait PropertiesTrait {
     }
     okVersions exists (javaVersion startsWith _)
   }
+
+  // provide a main method so version info can be obtained by running this
+  def main(args: Array[String]): Unit = {
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/config/Properties.scala]: This commit is called "Remove unnecessary main" but it adds the main back.

File: compiler/src/dotty/tools/dotc/config/CompilerCommand.scala [hunk 2]
```diff
@@ -115,6 +115,9 @@ object CompilerCommand extends DotClass {
     else if (settings.version.value) {
       ctx.echo(versionMsg)
       Nil
+    } else if (!Properties.isJavaAtLeast("1.8")) {
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/config/CompilerCommand.scala]: I don't think dotty is going to work out of the box on Java 9, so we should probably make this an equality check

Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/config/CompilerCommand.scala]: @DarkDimius The CI shows this line doesn't work.

File: compiler/src/dotty/tools/dotc/config/Properties.scala (hunk 1 above)
Comment (DarkDimius) [inline compiler/src/dotty/tools/dotc/config/Properties.scala]: Yes, something got screwed during rebase. Commit messages also got off-by-one, and the implementation of java8 check simply disappeared.

**Comment by DarkDimius** [comment]
re-rebased and force-pushed. Checked locally.

**Comment by liufengyun** [comment]
LGTM, thanks @DarkDimius !

**Pull request merged**